modules/remotecontrol/Controller.js

```diff
@@ -44,34 +47,113 @@ function getModifiers(event) {
+                        ConferenceEvents.ENDPOINT_MESSAGE_RECEIVED,
+                        permissionsReplyListener);
+                    resolve(result);
+                }
+            };
+            APP.conference.addConferenceListener(
+                ConferenceEvents.ENDPOINT_MESSAGE_RECEIVED,
+                permissionsReplyListener);
+            this.requestedParticipant = userId;
+            this._sendRemoteControlEvent(userId, {
+                type: EVENT_TYPES.permissions,
+                action: PERMISSIONS_ACTIONS.request
+            }, e => {
+                APP.conference.removeConferenceListener(
+                    ConferenceEvents.ENDPOINT_MESSAGE_RECEIVED,
+                    permissionsReplyListener);
+                this.requestedParticipant = null;
+                reject(e);
+            });
+        });
+    }
+
+    /**
+     * Handles the reply of the permissions request.
+     * @param {JitsiParticipant} participant the participant that has sent the
+     * reply
+     * @param {object} event the remote control event.
+     */
+    _handleReply(participant, event) {
+        const remoteControlEvent = event.event;
+        const userId = participant.getId();
+        if(this.enabled && event.type === REMOTE_CONTROL_EVENT_TYPE
+            && remoteControlEvent.type === EVENT_TYPES.permissions
+            && userId === this.requestedParticipant) {
+            if(remoteControlEvent.action === PERMISSIONS_ACTIONS.grant) {
+                this.controlledParticipant = userId;
+                this._start();
+                return true;
+            } else if(remoteControlEvent.action === PERMISSIONS_ACTIONS.deny) {
+                return false;
+            } else {
+                throw new Error("Unknown reply received!");
+            }
+        } else {
+            //different message type or another user -> ignoring the message
+            return null;
+        }
+    }
+
+    /**
+     * Handles remote control stopped.
+     * @param {JitsiParticipant} participant the participant that has sent the
+     * event
+     * @param {object} event the the remote control event.
+     */
+    _handleRemoteControlStoppedEvent(participant, event) {
+        if(this.enabled && event.type === REMOTE_CONTROL_EVENT_TYPE
+            && event.event.type === EVENT_TYPES.stop
+            && participant.getId() === this.controlledParticipant) {
+            this._stop();
+        }
     }
 
     /**
      * Starts processing the mouse and keyboard events.
-     * @param {JQuery.selector} area the selector which will be used for
-     * attaching the listeners on.
      */
-    start(area) {
+    _start() {
         if(!this.enabled)
             return;
-        this.area = area;
+        APP.conference.addConferenceListener(
+            ConferenceEvents.ENDPOINT_MESSAGE_RECEIVED,
+            this.stopListener);
+        this.area = $("#largeVideoWrapper");
```

Comment:
I don't really like how large video wrapper selector is hardcoded here. Would it be possible to get it somehow from the video layout or pass it as an argument ?